### PR TITLE
[DecimalV3](opt) opt the function of decimalv3 to_string logic

### PR DESCRIPTION
**Pull request by HappenLee**
# Proposed changes

After the opt:

```
mysql> select sum(length(cast (l_tax as string))) from lineitem;
+-----------------------------------------+
| sum(length(CAST(`l_tax` AS CHARACTER))) |
+-----------------------------------------+
|                              1680076256 |
+-----------------------------------------+
1 row in set (1 min 50.61 sec)

mysql> select sum(length(cast (l_tax as string))) from lineitem;
+-----------------------------------------+
| sum(length(CAST(`l_tax` AS CHARACTER))) |
+-----------------------------------------+
|                              1680076256 |
+-----------------------------------------+
1 row in set (2 min 29.14 sec)

mysql> select sum(length(cast (l_tax as string))) from lineitem;
+-----------------------------------------+
| sum(length(CAST(`l_tax` AS CHARACTER))) |
+-----------------------------------------+
|                              1680076256 |
+-----------------------------------------+
1 row in set (31.20 sec)

mysql> select sum(length(cast (l_tax as string))) from lineitem;
+-----------------------------------------+
| sum(length(CAST(`l_tax` AS CHARACTER))) |
+-----------------------------------------+
|                              1680076256 |
+-----------------------------------------+
1 row in set (30.47 sec)
```

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.29 seconds
 load time: 479 seconds
 storage size: 17170826662 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230206062917_clickbench_pr_90685.html

File: be/src/vec/core/types.h [hunk 1]
```diff
@@ -325,6 +341,44 @@ struct Decimal {
         return *this;
     }
 
+    std::string to_string(UInt32 scale) const {
+        static constexpr auto precision =
+                std::is_same_v<T, Int32> ? 9 : (std::is_same_v<T, Int64> ? 18 : 38);
```
Comment (Gabriel39) [inline be/src/vec/core/types.h:346]: Use `BeConsts::MAX_DECIMAL32_PRECISION / MAX_DECIMAL64_PRECISION / MAX_DECIMAL128_PRECISION` to replace `9 / 18 / 38`

File: be/src/vec/core/types.h [hunk 2]
```diff
@@ -325,6 +341,44 @@ struct Decimal {
...
+        int max_result_length = precision + (scale > 0) // Add a space for decimal place
+                                + (scale == precision)  // Add a space for leading 0
+                                + (is_nagetive);        // Add a space for negative sign
+        std::string str = std::string(max_result_length, '0');
+
+        T abs_value = value;
+        int pos = 0;
+
+        if (is_nagetive) {
+            abs_value = -value;
```
Comment (Gabriel39) [inline be/src/vec/core/types.h:357]: The result is unexpected if `value` is a min value of a signed numeric type

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**